src/cartridges/int_adyen_SFRA/cartridge/controllers/Adyen.js

```diff
@@ -186,118 +189,134 @@ server.post("Authorize3DS2", server.middleware.https, function (
-        URLUtils.url(
-          "Checkout-Begin",
-          "stage",
-          "payment",
-          "paymentError",
-          Resource.msg("error.payment.not.valid", "checkout", null)
-        )
-      );
-      return next();
-    }
 
-    const paymentDetailsRequest = {
-      paymentData: paymentInstrument.custom.adyenPaymentData,
-      details: details,
-    };
+      const result = adyenCheckout.doPaymentDetailsCall(paymentDetailsRequest);
 
-    const result = adyenCheckout.doPaymentDetailsCall(paymentDetailsRequest);
+      function checkAuthOrChallengeShopper() {
+        if (
+          (result.error || result.resultCode !== "Authorised") &&
+          result.resultCode !== "ChallengeShopper"
+        ) {
+          //Payment failed
+          Transaction.wrap(function () {
+            OrderMgr.failOrder(order, true);
+            paymentInstrument.custom.adyenPaymentData = null;
+          });
+          res.redirect(
+            URLUtils.url(
+              "Checkout-Begin",
+              "stage",
+              "payment",
+              "paymentError",
+              Resource.msg("error.payment.not.valid", "checkout", null)
+            )
+          );
+          return next();
+        } else if (result.resultCode === "ChallengeShopper") {
+          //Redirect to ChallengeShopper
+          res.redirect(
+            URLUtils.url(
+              "Adyen-Adyen3DS2",
+              "resultCode",
+              result.resultCode,
+              "token3ds2",
+              result.authentication["threeds2.challengeToken"]
+            )
+          );
+          return next();
+        }
+      }
+      checkAuthOrChallengeShopper();
 
-    if (
-      (result.error || result.resultCode !== "Authorised") &&
-      result.resultCode !== "ChallengeShopper"
-    ) {
-      //Payment failed
+      //delete paymentData from requests
       Transaction.wrap(function () {
-        OrderMgr.failOrder(order, true);
         paymentInstrument.custom.adyenPaymentData = null;
       });
-      res.redirect(
-        URLUtils.url(
-          "Checkout-Begin",
-          "stage",
-          "payment",
-          "paymentError",
-          Resource.msg("error.payment.not.valid", "checkout", null)
-        )
-      );
-      return next();
-    } else if (result.resultCode === "ChallengeShopper") {
-      //Redirect to ChallengeShopper
-      res.redirect(
-        URLUtils.url(
-          "Adyen-Adyen3DS2",
-          "resultCode",
-          result.resultCode,
-          "token3ds2",
-          result.authentication["threeds2.challengeToken"]
-        )
-      );
-      return next();
-    }
 
-    //delete paymentData from requests
-    Transaction.wrap(function () {
-      paymentInstrument.custom.adyenPaymentData = null;
-    });
+      //custom fraudDetection
+      const fraudDetectionStatus = { status: "success" };
 
-    //custom fraudDetection
-    const fraudDetectionStatus = { status: "success" };
+      // Places the order
```

Comment:
repetitive blocks of code like this one, we will refactor them in the other refactoring ticket right?